Branch-SDK/Branch-SDK/Requests/BranchOpenRequest.m

```diff
@@ -118,9 +118,8 @@ - (void)processResponse:(BNCServerResponse *)response error:(NSError *)error {
     }
     
     // Check if there is any Branch View to show
-    NSString *branchViewJsonString = data[BRANCH_RESPONSE_KEY_BRANCH_VIEW_DATA];
-    if(branchViewJsonString != nil && branchViewJsonString.length) {
-        NSDictionary *branchViewDict = [BNCEncodingUtils decodeJsonStringToDictionary:branchViewJsonString];
+    NSDictionary *branchViewDict = data[BRANCH_RESPONSE_KEY_BRANCH_VIEW_DATA];
+    if (branchViewDict != nil) {
```

Comment:
maybe to avoid crashes like the ones we saw recently you could make sure that it's actually a dictionary, not a string, number, etc. You can use isKindOfClass